src/improvements/doc/NEW_TEMPLATE_GUIDE.md

```diff
@@ -0,0 +1,273 @@
+# New Template Creation Guide
+
+This guide explains how to create new Solidity templates for the superchain task system. Templates are the foundation for standardizing and securing task execution across different networks.
+
+## Quick Start
+
+To scaffold a new template:
+
+```bash
+cd src/improvements/
+just new template
+```
+
+This will create a new Solidity file in `src/improvements/template/` with the basic structure required.
+
+## Template Structure
+
+### 1. Required Components
+
+Every template must implement the following functions:
```

Comment:
Guessing you're going to revisit this after OPCM changes get added. For example, `_build()` is going to change and be split across two functions iirc.